src/crawlee/http_clients/_httpx.py

```diff
@@ -222,13 +229,11 @@ def _get_client(self, proxy_url: str | None) -> httpx.AsyncClient:
         if proxy_url not in self._client_by_proxy_url:
             # Prepare a default kwargs for the new client.
             kwargs: dict[str, Any] = {
-                'transport': _HttpxTransport(
-                    proxy=proxy_url, http1=self._http1, http2=self._http2, verify=self._verify
-                ),
+                'transport': self._transport,
                 'proxy': proxy_url,
                 'http1': self._http1,
                 'http2': self._http2,
-                'verify': self._verify,
+                'verify': self._ssl_context,
```

Comment:
We should probably set the `transport` and `verify` fields *after* the `kwargs.update(self._async_client_kwargs)` line. I believe we don't want to let users override `transport` ever, and `verify` is not part of `self._async_client_kwargs`, but it'll be more obvious how we handle it that way.